### PR TITLE
add a demonstration task, illustrating reconciliation of missing events

### PR DESCRIPTION
**Pull request by isaacbowen**
Related discussion: https://usemechanic.slack.com/archives/CURE65U2W/p1632825406327200

Related documentation: https://learn.mechanic.dev/core/shopify/events/reconciling-missing-events

### CODE REVIEW

**Comment by isaacbowen** [comment]
@tekhaus I think you've addressed this kind of thing in your own work; anything you'd add/change here?

File: docs/demonstration-auto-tag-new-orders-with-reconciliation/script.liquid [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            }
+          }
+        }
+      {% endcapture %}
+  
+      {% assign result = result_json | parse_json %}
+    {% endif %}
+  
+    {% for order_edge in result.data.orders.edges %}
+      {% comment %}
```
Comment (tekhaus) [inline docs/demonstration-auto-tag-new-orders-with-reconciliation/script.liquid:73]: Is that possible in this context? Could this query return a record that should have been exclude by tag, while at the same time listing out that the order does in fact have the tag in question?

**Comment by tekhaus** [comment]
> @tekhaus I think you've addressed this kind of thing in your own work; anything you'd add/change here?

Interestingly, I have _not_ needed to have such an aggressive reconciliation set up in my work. Typically, where it's warranted, I have added a daily full scan run of whatever the underlying resource is, and the logic within the task would weed out the previously processed entities (often through a query filter as you have done here).

But this looks solid as a general solution for anyone needing that type of guarantee.

File: docs/demonstration-auto-tag-new-orders-with-reconciliation/script.liquid (hunk 1 above)
Comment (isaacbowen) [reply docs/demonstration-auto-tag-new-orders-with-reconciliation/script.liquid:73]: Yeah, it came up in my testing. Shopify's API for this stuff is backed by a search index of some kind, and (at least on the dev store I used here) there was some lag between the moment the order was tagged, and the moment when the search results started responding to that tag. Less than 10 seconds, but it was still there.

**Pull request merged**